jenkins/format-submission.py

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env python3
+
+import argparse
+import json
+import requests
+from requests.auth import HTTPBasicAuth
+from requests.exceptions import HTTPError
+import subprocess
+import zipfile
+
+# submission process starts from MICROSUB project on issues.citrite.net
+HOST = "https://issues.citrite.net"
+
+# Submission issue type has multiple custom fields
+SUPPORT_URL = "customfield_12635"
+DOCUMENT_URL = "customfield_31030"
+PRIVACY_URL = "customfield_31032"
+TERMS_URL = "customfield_31031"
+VENDOR = "customfield_31230"
+
+# metadata for bundle is held in the METADATA_FILE
+METADATA_FILE = "metadata.json"
+
+# parse args to obtain credentials for Jira API
+parser = argparse.ArgumentParser(description='Format the export to include extra metadata from third parties')
+parser.add_argument('--issueId', type=str, required=True, help='id of the issue that triggered the build')
+parser.add_argument('--svcacctPwd', type=str, required=True, help='password for the service account making call to Jira')
+parser.add_argument('--svcacctName', type=str, required=True, help='service account name used for Jira API calls')
+args = parser.parse_args()
+SVCACCT_NAME = args.svcacctName
+SVCACCT_PWD = args.svcacctPwd
+ISSUE_ID = args.issueId
+AUTH = HTTPBasicAuth(SVCACCT_NAME, SVCACCT_PWD)
+HEADERS = {'Content-Type': 'application/json'}
+
+
+def _error_writeback(error):
+    errorMessage = 'Jenkins script failed: {}'.format(error)
+    payload = json.dumps( {
+        "body": errorMessage
+    } )
+    try:
+        response = requests.post(
+            url='{}/rest/api/2/issue/{}/comment'.format(HOST, ISSUE_ID),
+            headers=HEADERS,
+            auth=AUTH,
+            data=payload
+        )
+        response.raise_for_status()
+    except requests.exceptions.RequestException as e:
+        print('failed to write error to Jira: {}'.format(e))
+    exit(1)
+
+
+def get_mapp_file(issueJson):
+    # only allow one attachment and must be a .mapp file
+    mappFile = ""
+    numAttachments = len(issueJson['fields']['attachment'])
+    if numAttachments > 1:
+        _error_writeback("only attach a single file")
```

Comment:
Should expand these writeback comments so it is clearer what needs to be done next. Can they just remove the extra files and this process will automatically restart? or is something more required?